### ISSUE TITLE
Add rewrite log for when rewriting is enabled

### ISSUE THREAD

**Issue by bdw429s** [cfmlprojects/runwar#114]
I like the basic rewrite info stuff that goes into the out logs, but we've lost all debug and trace logging in Tuckey which is no good at all.  Let's create a new rewrite log when rewriting is enabled that will also capture all logs from Tuckey.  When debug and trace logging are enabled, Tuckey's debug and trace logs will go there.

**Comment by denuno** [comment]
We can _kind_ of do this by directing the undertow servlet logger to a file appender we want to see the rewrite log entries in, but for some reason thus far I haven't been able to get the tuckey logger to not be a child of the servlet logger, so it'd be other stuff too-- that is, eventually it could get it's own log, but for the next release at least the most we could do would be add a `servlet.log` or some such that would _mostly_ just be the rewrite stuff, but also whatever else gets caught in there.

**Comment by bdw429s** [comment]
Hmm, I just posted about this on another ticket where the Tuckey logs seem to be getting nested inside the servlet logs, and their logging categories are being overridden to.

```
[INFO ] io.undertow.servlet: org.tuckey.web.filters.urlrewrite.Condition TRACE: outcome false
```

Should I ask over on the undertow list about it?

**Comment by denuno** [comment]
I guess it wouldn't hurt.  Pretty sure it's been that way for a while, and I just manually replaced the `io.undertow.servlet: ` bit before, IIRC.  Really I'd like to switch to undertow rewrites, I bet they'd be faster, but that'd take writing some type of adapter to load the configs, which isn't trivial.

**Comment by bdw429s** [comment]
@denuno Thread started here:
http://lists.jboss.org/pipermail/undertow-dev/2018-April/002196.html

**Comment by denuno** [comment]
Thanks for asking about that, it led me to passing in `SLF4J` as the log level to tuckey, which lets it use the loggers... I didn't bother tweaking the tuckey fork as now the logging config is in code vs. config files so 'tis easier to manhandle. :smiley: 

The new param is `--urlrewrite-log /path/to/log/file`, and there are like 4 loggers for those rewrites, so I dunno... I think you were saying you only wanted to see one line or some such, perhaps take a look at the output and let me know which logger and what level we want to output.

Basically I'm thinking `--debug` will just do that one line or whatnot, and then `-level TRACE` will do the firehose... could also add a `--urlrewrite-log-level` param I guess, so it's separate from the rest (the log file content should be separate, but the console will have all the other trace stuff too), meh, *shrugles*.

**Comment by bdw429s** [comment]
Here is the line of debug that I'd like to see in the regular old console output when `--debug` is set:

```
[DEBUG] org.tuckey.web.filters.urlrewrite.RuleExecutionOutput: needs to be forwarded to /index.cfm/Main
```
So, that's the `DEBUG` messages from the `org.tuckey.web.filters.urlrewrite.RuleExecutionOutput` logging category.

It basically just lets me know that the URL was rewritten and what it was rewritten to.  I'd also like to have a single line before that in debug mode that simply shows what the incoming request was for.  Which in the case above, would have been `/Main`.  Right now it seems all request-specific logging is gone in debug mode.  

I'll test out the new rewrite logs.  As far as logging levels, I'm totally ok with the rewrite levels matching the other logging levels which is pretty much debug and trace.  I don't think Tuckey really has many warn or error logs, but those are always ok in the logs regardless of level since they indicate an issue of some sort.  info and debug can show up when debug is on and then trace can be added in once trace is enabled.

**Comment by bdw429s** [comment]
The logs are starting to look a lot better BTW.  How that those tuckey log entries aren't nested inside of the undertow categories it really cleaned up the output a bit

**Comment by bdw429s** [comment]
> The new param is --urlrewrite-log /path/to/log/file,

Just confirming, the other logs have several CLI options to set the directory, file name, etc.  Is the rewrite log a single shot thing?  I'm totally cool with this as I'm not exposing all the bits and pieces to CommandBox users right now anyway.  Just wanted to check since it seemed inconsistent.

**Comment by bdw429s** [comment]
The rewrite log looks fantastic!  There's nothing logged there for a "normal" start, it gets debug and info messages for a debug start and trace messages added in for a trace start.  That is perfect.  I think the only two remaining things to think about are:
1) Can we get that `org.tuckey.web.filters.urlrewrite.RuleExecutionOutput` category added to the console out log when debug logging is on just to show that a rewrite fired?
2) Should we now remove the REST of the rewrite logs from the standard console logging?  I'm tempted to say yes.  It's one of those things people only care about if they're troubleshooting their rewrites and now that we have a separate log for it, we can clean up the console log a bunch.  Turning on the rewrite log will be easy to do and I'll even add a nice convenience command to tail it as well for quick debugging .
